goat-sdk/goat
# feat(plugins): add mode governance plugin

# Relates to:
Hackthon issue Add Mode Governance Plugin Examples

# Background
## What does this PR do?
This PR adds a new example showcasing the Mode Governance plugin integration with the GOAT SDK using Vercel AI. The example demonstrates how to:
- Set up and initialize the Mode Governance plugin
- Interact with Mode DAO governance functionality
- Execute governance actions using GPT-4 and GOAT tools
- Handle voting power management and gauge voting

## What kind of change is this?
- ✨ New feature (adds new example code)
- 📝 Documentation (includes example documentation)

# Documentation changes needed?
My changes include the necessary documentation within the example:
- Added comprehensive README.md with setup instructions
- Included inline code comments explaining key functionality
- Added example prompts and usage patterns

# Testing
## Detailed testing steps
1. Navigate to examples/vercel-ai/mode-governance
2. Copy .env.template to .env and configure:
   - Set OPENAI_API_KEY
   - Set WALLET_PRIVATE_KEY (with Mode tokens)
   - Set RPC_PROVIDER_URL
3. Run `npm install`
4. Execute `npm start`
5. Verify the following functionalities:
   - Successful connection to Mode Network
   - Correct retrieval of voting power
   - Ability to view available gauges
   - Successful vote casting
   - Accurate retrieval of voting history

## For plugins
- [x] I have tested this change locally with hosted wallets (Crossmint Smart Wallets)

## Discord username
codingshthefounder

## Telegram username 
codingsh
